generator/sbpg/targets/resources/MessageTest.java.j2

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (C) 2015-2018 Swift Navigation Inc.
+ * Contact: https://support.swiftnav.com
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+// This file was auto-generated from (((s.src_filename))) by generate.py. Do not modify by hand!
+
+import java.math.BigInteger;
+
+import org.junit.Test;
+
+import org.json.JSONObject;
+
+import com.swiftnav.sbp.SBPMessage;
+((* set names = s.tests|map(attribute='msg')|map(attribute='name')|unique|list *))
+((*- for name in names *))
+import com.swiftnav.(((s.package))).(((name)));
+((* endfor *))
+
+
+((*- macro compare_value(prefix, value) *))
+((*- if value is string_type *))
+        org.junit.Assert.assertEquals(msg.(((prefix))), (((value|str_escape))) );
+((*- elif value is array_type *))
+((*- for ff in value *))((( compare_value( (((prefix))) + '[' + (((loop.index0|to_str))) + ']', (((ff))) ) )))((*- endfor *))
+((*- elif value is dict_type *))
+((*- for k in (((value|sorted))) *))((( compare_value( (((prefix))) + '.' + (((k))), (((value[k]))) ) )))((*- endfor *))
+((*- elif value is float_type *))((=
+    Note: the ("%.17e"|format(value)) filter is intended to preserve float
+    literal precision accross all value ranges. =))
+        org.junit.Assert.assertEquals(msg.(((prefix))), ((("%.17e"|format(value)))), DELTA);
+((*- elif value in [2159176030, 3411995557, 7909447587, 8409447265, 8907446923, 9406446591, 9906446269] *))
```

Comment:
These seven are somehow different, possibly a bug in the parsing.

Reply:
These all look larger than what will fit in 32-bit unsigned int-- 🤷‍♀️

Reply:
The first two were erroneously listed here; and the rest are all greater than unsigned int.
However the problem was endianness. It is only a few attributes where this crops up, and it silently failed, so not surprising nobody noticed this.